sanic/sessions.py

```diff
@@ -0,0 +1,303 @@
+# -*- encoding: utf-8 -*-
+
+import hashlib
+
+from itsdangerous import URLSafeTimedSerializer
+from datetime import datetime
+
+import ujson as json
+from sanic.exceptions import BadSignature
+
+
+def total_seconds(td):
+    """Returns the total seconds from a timedelta object.
+    :param timedelta td: the timedelta to be converted in seconds
+    :returns: number of seconds
+    :rtype: int
+    """
+    return td.days * 60 * 60 * 24 + td.seconds
+
+
+class SessionMixin(object):
```

Comment:
Yes this interface is 90% from flask source, I just made minimum modify to work with sanic. Why bother build a whole new thing if there's already one. 

Extension will definitely work as same, I just thought it would be nice to have it built in.